src/react-components/home-root.js

```diff
@@ -54,13 +58,33 @@ class HomeRoot extends Component {
     }
   };
 
-  showDialog = dialogType => {
-    return e => {
-      e.preventDefault();
-      e.stopPropagation();
-      this.setState({ dialogType });
-    };
-  };
+  closeDialog() {
+    this.setState({ dialog: null });
+  }
+
+  showSlackDialog() {
+    this.setState({ dialog: <SlackDialog onClose={this.closeDialog} /> });
```

Comment:
Ah interesting, I was expecting you migh have ended up doing something like:
```
const Dialog = this.props.dialog
<Dialog onClose={this.closeDialog| />
```
In render. A bit less ceremony, but the way you have done it lets you vary the arguments which seems to be required.